device/device.yaml

```diff
@@ -0,0 +1,49 @@
+components:
+  schemas:
+    Device:
+      description: >-
+        Device.
+      type: object
+      required: [name]
+      properties:
+        name:

```

Comment:
We need to make clear differentiation between device names in the inventory (and possibly performing subgraph search using names) and logical IDs of the devices to be used in the testbed definition/reservation as well as within the test. From this perspective, would this `name` attribute be used for the search or as a logical ID? I propose to call the attribute for logical IDs as `id`, and those used for search as `name`

Reply:
`name` is any logical name and no need to represent same from inventory, Are you suggesting to change the attribute from `name` to `id`?

Reply:
Yes, because in the inventory we have device and interface names, why would we handle them differently than a vendor or model? I think we should consider that someone might want to use a regex to match device names. I think `id` would work for the logical name, in fact, this is what ondatra is using: https://github.com/openconfig/ondatra/blob/main/proto/testbed.proto#L34

Reply:
changed `name` to `id`

Reply:
We need to have `name` as an option for device filtering. This way a user can request a specific device from the lab, or maybe via a regex search. In other words, we need `id` to uniquely identify logical devices, and `name` as another parameter for the search.

Reply:
We have attributes right, if user wants to match the name from inventory user can use something like this right.

```devices:
- attributes:
  - key: name
    value: name_in_inventory

Reply:
added `name` attribute.